File: Frontend/src/domain/valueObject/dueDate.ts
export class DueDate {
    private readonly value: Date | null
    private constructor(input:  Date | null ) {
            this.value= input
    }

    static NewDueDate(future: Date | null): DueDate {
        if (future && future < new Date()) {
            throw new Error("期日は現在以降の日付にしてください")
        } else {
            return new DueDate(future)
        }
    }

    static FromExisting(date: Date | null): DueDate {
        return new DueDate(date)
    }

    Value() {
        return this.value
    }
}

File: GoAPI/src/domain/valueobject/DueDate.go
package valueobject

import (
	"fmt"
	"time"
)

type DueDate struct {
	dueDate time.Time
}

func NewDueDate(input *time.Time) (DueDate, error) {
	if input.Before(time.Now()) {
		return DueDate{}, fmt.Errorf("過去の日時は入力できません")
	}
	return DueDate{dueDate: *input}, nil
}

func ExistingDueDate(input *time.Time) DueDate {
	return DueDate{dueDate: *input}
}

func (d DueDate) Value() time.Time {
	return d.dueDate
}
